### ISSUE TITLE
Gradle multi-project build hangs in "CONFIGURING" state when external properties file is missing

### ISSUE THREAD

**Issue by dcmoeller** [sidney-simmons/gradle-plugin-external-properties#25]
My current task is to split up a "simple" (= "non-multi-project") Gradle build to what Gradle calls "multi-project" build.
In the former "simple" case Gradle build runs even if (e.g. in CI environment) configured external properties file `gradle-local.properties` is missing.
But with "multi-project" build, Gradle (version 6.5.1) "hangs" in `CONFIGURING` state!

To get more details, I've executed Gradle build with `--debug` flag to see this output (shortened here):

```
$ ./gradlew --no-daemon --debug clean
[...]
[INFO] [org.gradle.api.Project] Cannot load external properties file. Continuing gracefully. file = /home/[...]/gradle-local.properties
[LIFECYCLE] [org.gradle.cache.internal.DefaultFileLockManager] 
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
[DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
<========-----> 66% CONFIGURING [33s]
^C
```

I've stopped execution with `Ctrl-C`.
`build.gradle` of root project contains:

```
plugins {
  id 'com.sidneysimmons.gradle-plugin-external-properties' version '1.0.0' apply false
}
```

Subproject's `build.gradle`:

```
apply plugin: 'com.sidneysimmons.gradle-plugin-external-properties'
externalProperties {
  resolver file("${rootDir}/gradle-local.properties")
}
```

Maybe you can tell more about root cause of this problem and how to fix it.

Regards

Christian

**Comment by dcmoeller** [comment]
Same result with missing default external properties file when removing code this code block:

**Comment by sidney-simmons** [comment]
Hey Christian I appreciate all the details you provided!  I have not seen this before.  Let me try to reproduce locally.

**Comment by sidney-simmons** [comment]
@dcmoeller so it looks like the traversal up through the parent projects was broken.  Instead of getting the parent from the project in the current loop iteration we were getting the parent from the main project.  So we started an infinite loop instead of traversing up to the root project.  🤦‍♂️ 

I believe I fixed it (also added some extra logging) via #26.  I pushed this change up to the gradle plugin portal as version `1.0.1`.  Let me know if this solves it on your end.
